src/main/java/de/tum/in/www1/artemis/service/StudentExamService.java

```diff
@@ -248,6 +260,62 @@ else if (exercise instanceof ModelingExercise) {
         }
     }
 
+    /**
+     * Automatically assess the modeling- and programming exercises of student exams of an exam which are not submitted with 0 points.
+     * The assessment is counted as {@link AssessmentType#SEMI_AUTOMATIC} to make sure it is not considered for manual assessment,
+     * see {@link StudentParticipationRepository#findByExerciseIdWithLatestSubmissionWithoutManualResultsAndNoTestRunParticipation}.
+     * The assessor is artemis_admin.
+     *
+     * @param examId the exam id
+     * @return returns the number of assessedSubmissions
+     */
+    public int assessUnsubmittedStudentExams(final Long examId) {
+        int numberOfAssessedSubmissions = 0;
+        User artemisAdmin = userService.getUserWithGroupsAndAuthorities("artemis_admin");
```

Comment:
Should it be better to define `artemis_admin` as a constant so that it's easier to modify in the future?

Reply:
This is a bad choice in my opinion. We cannot assume that this user exists and it also does not make sense to set it as assessor. Please change it to the currently logged in user, i.e. the user who clicks the button.

Reply:
I reverted to using the assessor who clicks on the button